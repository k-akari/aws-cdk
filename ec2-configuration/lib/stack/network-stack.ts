import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { Vpc } from 'aws-cdk-lib/aws-ec2';
import { VPC } from '../resource/vpc';
import { Subnet } from '../resource/subnet';
import { InternetGateway } from '../resource/internet-gateway';
import { RouteTable } from '../resource/route-table';
import { NetworkAcl } from '../resource/network-acl';

export class NetworkStack extends Stack {
  public readonly vpc: Vpc;
  public readonly subnet: Subnet;

  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);
   
    const vpc = new VPC(this);
    this.vpc = vpc.vpc;
    this.subnet = new Subnet(this, this.vpc);
    const internetGateway = new InternetGateway(this, this.vpc);
    new RouteTable(this, this.vpc, this.subnet, internetGateway.igw);
    new NetworkAcl(this, this.vpc, this.subnet);
  }
}
