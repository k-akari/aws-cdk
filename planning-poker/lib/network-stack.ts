import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { Vpc } from './resource/vpc';
import { Subnet } from './resource/subnet';
import { InternetGateway } from './resource/internet-gateway';
import { ElasticIp } from './resource/elastic-ip';
import { NatGateway } from './resource/nat-gateway';
import { RouteTable } from './resource/route-table';
import { NetworkAcl } from './resource/network-acl';

export class NetworkStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);
   
    const vpc = new Vpc(this);
    const subnet = new Subnet(this, vpc);
    const internetGateway = new InternetGateway(this, vpc);
    const elasticIp = new ElasticIp(this);
    const natGateway = new NatGateway(this, subnet, elasticIp);
    new RouteTable(this, vpc, subnet, internetGateway, natGateway);
    new NetworkAcl(this, vpc, subnet);
  }
}
