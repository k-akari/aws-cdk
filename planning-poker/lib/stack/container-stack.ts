import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { Vpc } from 'aws-cdk-lib/aws-ec2';
import { IamRole } from '../resource/iam-role';
import { Ecr } from '../resource/ecr';
import { Ecs } from '../resource/ecs';

export class ContainerStack extends Stack {
  constructor(scope: Construct, id: string, vpc: Vpc, iamRole: IamRole, props?: StackProps) {
    super(scope, id, props);
   
    new Ecr(this);
    new Ecs(this, vpc, iamRole);
  }
}
