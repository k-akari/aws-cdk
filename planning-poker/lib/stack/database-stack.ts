import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { Subnet } from '../resource/subnet';
import { Rds } from '../resource/rds';

export class DatabaseStack extends Stack {
  constructor(scope: Construct, id: string, subnet: Subnet, props?: StackProps) {
    super(scope, id, props);
   
    new Rds(this, subnet);
  }
}
