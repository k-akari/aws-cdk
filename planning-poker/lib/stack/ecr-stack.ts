import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { Ecr } from '../resource/ecr';

export class EcrStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);
   
    new Ecr(this);
  }
}
