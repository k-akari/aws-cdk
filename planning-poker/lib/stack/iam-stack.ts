import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { IamRole } from '../resource/iam-role';

export class IamStack extends Stack {
  public readonly iamRole: IamRole;

  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);
   
    this.iamRole = new IamRole(this);
  }
}
