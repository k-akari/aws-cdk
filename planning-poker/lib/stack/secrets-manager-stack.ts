import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { SecretsManager } from '../resource/secrets-manager';

export class SecretsManagerStack extends Stack {
  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);
   
    new SecretsManager(this);
  }
}
