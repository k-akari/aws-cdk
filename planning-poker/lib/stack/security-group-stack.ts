import { Stack, StackProps } from 'aws-cdk-lib';
import { Construct } from 'constructs';
import { Vpc } from '../resource/vpc';
import { SecurityGroup } from '../resource/security-group';

export class SecurityGroupStack extends Stack {
  constructor(scope: Construct, id: string, vpc: Vpc, props?: StackProps) {
    super(scope, id, props);

    // Security Group
    const securityGroup = new SecurityGroup(vpc);
    securityGroup.createResources(this);
  }
}
