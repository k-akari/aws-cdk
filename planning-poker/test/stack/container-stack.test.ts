import { App } from 'aws-cdk-lib';
import { Template } from 'aws-cdk-lib/assertions';
import * as Network from '../../lib/stack/network-stack';
import * as Iam from '../../lib/stack/iam-stack';
import * as Container from '../../lib/stack/container-stack';

test('ContainerStackSnapshot', () => {
  const serviceName = 'service';
  const envType = 'test';
  const app = new App({
    context: {
      'serviceName': serviceName,
      'envType': envType
    }
  });
  const networkStack = new Network.NetworkStack(app, 'NetworkStack');
  const iamStack = new Iam.IamStack(app, 'IamStack');
  const containerStack = new Container.ContainerStack(app, 'ContainerStack', networkStack.vpc, iamStack.iamRole);
  const template = Template.fromStack(containerStack);

  expect(template).toMatchSnapshot();
});
