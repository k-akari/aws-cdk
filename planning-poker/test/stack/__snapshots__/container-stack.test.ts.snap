// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContainerStackSnapshot 1`] = `
Object {
  "Parameters": Object {
    "BootstrapVersion": Object {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": Object {
    "AppF1B96344": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "ImageTagMutability": "IMMUTABLE",
        "RepositoryName": "service-test-app",
      },
      "Type": "AWS::ECR::Repository",
      "UpdateReplacePolicy": "Retain",
    },
    "EcsCluster97242B84": Object {
      "Properties": Object {
        "ClusterName": "service-test-ecs-cluster",
        "ClusterSettings": Array [
          Object {
            "Name": "containerInsights",
            "Value": "enabled",
          },
        ],
      },
      "Type": "AWS::ECS::Cluster",
    },
    "ServiceTaskDefinitionEFF764B2": Object {
      "Properties": Object {
        "Cpu": "256",
        "ExecutionRoleArn": Object {
          "Fn::ImportValue": "IamStack:ExportsOutputFnGetAttEcsTaskRole8DFA0181Arn43EC6E0B",
        },
        "Family": "service-test-task-definition",
        "Memory": "2048",
        "NetworkMode": "awsvpc",
        "RequiresCompatibilities": Array [
          "FARGATE",
        ],
        "TaskRoleArn": Object {
          "Fn::ImportValue": "IamStack:ExportsOutputFnGetAttEcsTaskExecutionRoleC0B6A84DArnF61B3BFD",
        },
      },
      "Type": "AWS::ECS::TaskDefinition",
    },
    "Web3C8945DB": Object {
      "DeletionPolicy": "Retain",
      "Properties": Object {
        "ImageTagMutability": "IMMUTABLE",
        "RepositoryName": "service-test-web",
      },
      "Type": "AWS::ECR::Repository",
      "UpdateReplacePolicy": "Retain",
    },
  },
  "Rules": Object {
    "CheckBootstrapVersion": Object {
      "Assertions": Array [
        Object {
          "Assert": Object {
            "Fn::Not": Array [
              Object {
                "Fn::Contains": Array [
                  Array [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  Object {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
